solved_exercises/dani/04-functions.py

```diff
@@ -7,29 +9,27 @@ def factorial(n):
     return result
 
 
-value = int(input("Please enter a numeric value:\n"))
-print("Factorial of", value, "is", factorial(value))
-
 # Write a function that flattens a list.
 def flatten(raw_list):
     flat_list = []
-    for element in raw_list:
-        if type(element) == list:
-            flat_list.extend(flatten(element))
-        else:
-            flat_list.append(element)
+    if not isinstance(raw_list, list):
+        flat_list = []
+    else:
+        for element in raw_list:
+            if type(element) == list:
```

Comment:
let's switch to ``isinstance`` check here too.

Reply:
done